manifests/params.pp

```diff
@@ -186,13 +186,16 @@
   $tftp_syslinux_files = undef
 
   # DHCP settings - requires optional DHCP puppet module
-  $dhcp                   = false
-  $dhcp_listen_on         = 'https'
-  $dhcp_managed           = true
-  $dhcp_interface         = 'eth0'
-  $dhcp_gateway           = '192.168.100.1'
-  $dhcp_range             = false
-  $dhcp_option_domain     = [$::domain]
+  $dhcp                    = false
+  $dhcp_split_config_files = true # smart-proxy 1.11+
+  $dhcp_listen_on          = 'https'
+  $dhcp_managed            = true
+  $dhcp_provider           = 'isc'
```

Comment:
My only nagging thought is that it prevents the provider being set with a plugin name that doesn't start with `dhcp_` or `dns_`, but I can only hope nobody does that.  I thought about making `dhcp_provider` work a bit more flexibly, but we're already committed with the DNS parameter.